mentat/feature_filters/truncate_filter.py

```diff
@@ -0,0 +1,45 @@
+from mentat.code_feature import CodeFeature, CodeMessageLevel
+from mentat.feature_filters.feature_filter import FeatureFilter
+
+
+class TruncateFilter(FeatureFilter):
+    def __init__(
+        self,
+        max_tokens: int,
+        model: str = "gpt-4",
+        code_map: bool = False,
+        respect_user_include: bool = True,
+    ):
+        self.max_tokens = max_tokens
+        self.model = model
+        self.code_map = code_map
+        self.respect_user_include = respect_user_include
+
+    async def filter(
+        self,
+        features: list[CodeFeature],
+    ) -> list[CodeFeature]:
+        """Truncate the features to max_token tokens."""
+        levels = [CodeMessageLevel.FILE_NAME]
```

Comment:
I would keep `levels` as a kwarg and move it to __init__. There are times, e.g. at the end of LLMFeatureFilter, where we'd want to truncate without modifying the levels.